bgpd/bgp_label.c

```diff
@@ -57,7 +57,7 @@ int bgp_parse_fec_update(void)
 	memset(&p, 0, sizeof(struct prefix));
 	p.family = stream_getw(s);
 	p.prefixlen = stream_getc(s);
-	stream_get(&p.u.prefix, s, PSIZE(p.prefixlen));
+	stream_get(p.u.val, s, PSIZE(p.prefixlen));
```

Comment:
it would be better to reference prefix part of the union - is there a reason why you're not?

Reply:
hi, the reason is that using the 'prefix' element of the union you need to remember that are other elements larger than it so the overflow never happens while using the exact container, or the biggest suitable container you can be sure by construction (i.e. you have space because the element used has always enough space no matter the other elements of the union)

Reply:
I think it's most intuitive to have the referenced field match the field and length being read.

Reply:
I agree, in the case of knowing the type beforehand (e.g. known IPv4 or IPv6). However, in this case the length is variable, so using the .prefix (1-byte container) element address is less "right" than using the .val (16-byte container) element. IMO, it would be better to have had just prefix[16] and val[16] would be unnecessary, but... :-)